FirebaseStorage/Tests/ObjCIntegration/ObjCPPAPITests.mm

```diff
@@ -0,0 +1,224 @@
+  storage.maxUploadRetryTime = storage.maxUploadRetryTime + 1;
+  [storage setMaxDownloadRetryTime:[storage maxDownloadRetryTime]];
+  storage.maxDownloadRetryTime = storage.maxDownloadRetryTime + 1;
+  [storage setMaxOperationRetryTime:[storage maxOperationRetryTime]];
+  [storage setCallbackQueue:[storage callbackQueue]];
+  FIRStorageReference *ref = [storage reference];
+  ref = [storage referenceForURL:@"my-url"];
+  ref = [storage referenceWithPath:@"my-path"];
+  [storage useEmulatorWithHost:@"my-host" port:123];
+}
+
+- (void)FIRStorageReferenceApis {
+  FIRStorageReference *ref = [[FIRStorage storage] reference];
+  [ref storage];
+  [ref bucket];
+  [ref fullPath];
+  [ref name];
+
+  ref = [ref root];
+  ref = [ref parent];
+  ref = [ref child:@"path"];
+
+  NSData *data = [NSData data];
+  FIRStorageUploadTask *task = [ref putData:data];
+  task = [ref putData:data metadata:nil];
+  task = [ref putData:data
+             metadata:nil
+           completion:^(FIRStorageMetadata *_Nullable metadata, NSError *_Nullable error){
+           }];
+
+  NSURL *file = [NSURL URLWithString:@"my-url"];
+  task = [ref putFile:file];
+  task = [ref putFile:file metadata:nil];
+  task = [ref putFile:file
+             metadata:nil
+           completion:^(FIRStorageMetadata *_Nullable metadata, NSError *_Nullable error){
+           }];
+
+  FIRStorageDownloadTask *task2 =
+      [ref dataWithMaxSize:123
+                completion:^(NSData *_Nullable data, NSError *_Nullable error){
+                }];
+
+  [ref downloadURLWithCompletion:^(NSURL *_Nullable URL, NSError *_Nullable error){
+  }];
+
+  task2 = [ref writeToFile:file];
+  task2 = [ref writeToFile:file
+                completion:^(NSURL *_Nullable URL, NSError *_Nullable error){
+                }];
+
+  [ref listAllWithCompletion:^(FIRStorageListResult *_Nonnull result, NSError *_Nullable error){
+  }];
+
+  [ref listWithMaxResults:123
+               completion:^(FIRStorageListResult *_Nonnull result, NSError *_Nullable error){
+               }];
+  [ref listWithMaxResults:123
+                pageToken:@"my token"
+               completion:^(FIRStorageListResult *_Nonnull result, NSError *_Nullable error){
+               }];
+
+  [ref metadataWithCompletion:^(FIRStorageMetadata *_Nullable metadata, NSError *_Nullable error) {
+    [ref updateMetadata:metadata
+             completion:^(FIRStorageMetadata *_Nullable metadata, NSError *_Nullable error){
+             }];
+  }];
+
+  [ref deleteWithCompletion:^(NSError *_Nullable error){
+  }];
+}
+
+#ifdef COCOAPODS
+- (void)FIRStorageConstantsTypedefs {
+  FIRStorageHandle __unused handle;
+  FIRStorageVoidDataError __unused funcPtr1;
+  FIRStorageVoidError __unused funcPtr2;
+  FIRStorageVoidMetadata __unused funcPtr3;
+  FIRStorageVoidMetadataError __unused funcPtr4;
+  FIRStorageVoidSnapshot __unused funcPtr5;
+  FIRStorageVoidURLError __unused funcPtr6;
+}
+#endif
```

Comment:
CocoaPods only?

Reply:
Yes, there's no way to make public typedefs in SwiftPM

Reply:
Ah right. I guess these tests won't run in SwiftPM though right?

Reply:
Right. The `#ifdef` enables the rest of the file to run in SwiftPM

Reply:
We're excluding the file entirely in SwiftPM, so I think it'd be fine to just eliminate the CocoaPods check but I'm fine leaving it for when we can eventually get support 

https://github.com/firebase/firebase-ios-sdk/blob/8623b798cbabaecedadb6321b6dad2c11d67d450/Package.swift#L1110-L1113